math/vecops/inc/ROOT/RVec.hxx

```diff
@@ -3242,6 +3242,277 @@ RVec<typename RVec<T>::size_type> Enumerate(const RVec<T> &v)
+ * \tparam Ret_t Return type used for arithmetic, which, if not explicitly specified
+ * in the template, is \p T if that is a floating point type, or double otherwise.
+ *
+ * \param start The first value in the sequence.
+ * \param end The last value in the sequence if \p endpoint is true; otherwise, \p end is excluded.
+ * \param n The number of evenly spaced entries to produce. The default value is 128, which is different than numpy's default value of 50.
+ * \param endpoint If true (default), \p end is included as the final element; if false, \p end is excluded.
+ *
+ * \return A vector (RVec<Ret_t>) containing \p n evenly spaced values.
+ *
+ * \note If \p n is 1, the resulting vector will contain only the value \p start.
+ * \note The check `if (!n || (n > std::numeric_limits<long long>::max()))` is used to ensure that:
+ *   - division by zero is avoided when calculating `step`
+ *   - n does not exceed std::numeric_limits<long long>::max(), which would indicate that a negative range (or other arithmetic issue)
+ *     has resulted in an extremely large unsigned value, thereby preventing an attempt to reserve an absurd
+ *     amount of memory.
+ * \note If the template parameter \c Ret_t is explicitly overridden with an integral type, the arithmetic may cause rounding issues. Consequently, the resulting sequence may not end exactly at \p end. To ensure that the sequence ends exactly at \p end, consider casting the result as follows: `RVec<integral_type>(Linspace(...))`. This behavior is different than NumPy in Python.
+ *
+ * \par C++23 Enumerate Support:
+ * With C++23, you can use the range-based enumerate view to iterate over the resulting vector with both the index
+ * and the value, similar to Python's `enumerate`. For example:
+ * ~~~{.cpp}
+ * for (auto const [index, val] : std::views::enumerate(ROOT::VecOps::Linspace(6, 10, 16))) {
+ *     // Process index and val.
+ * }
+ * ~~~
+ *
+ * \par Example code, at the ROOT prompt:
+ * ~~~{.cpp}
+ * using namespace ROOT::VecOps;
+ * cout << Linspace(-1, 5, 5) << "\n";
+ * // { -1, 0.5, 2, 3.5, 5 }
+ * cout << Linspace(3, 12, 5) << "\n";
+ * // { 3, 5.25, 7.5, 9.75, 12 }
+ * cout << Linspace(3, 12, 5, false) << "\n";
+ * // { 3, 4.8, 6.6, 8.4, 10.2 }
+ * Linspace<int, int>(1, 10, 3) << "\n";
+ * // { 1, 5, 9 }
+ * ~~~
+ */
+template <typename T = double, typename Ret_t = std::conditional_t<std::is_floating_point_v<T>, T, double>>
+inline RVec<Ret_t> Linspace(T start, T end, unsigned long long n = 128, const bool endpoint = true)
+{
+    if (!n || (n > std::numeric_limits<long long>::max())) // Check for invalid or absurd n.
+    {
+        return RVec<Ret_t>{};
+    }
+    
+    long double step = std::is_floating_point_v<Ret_t> ?
+    (static_cast<Ret_t>(end) - static_cast<Ret_t>(start)) / static_cast<Ret_t>(n - endpoint) :
+    static_cast<long double>(end - start) / (n - endpoint);
+        
+    RVec<Ret_t> temp(n);
+    temp[0] = static_cast<Ret_t>(start);
```

Comment:
```suggestion
    temp[0] = std::is_floating_point_v<Ret_t> ? static_cast<Ret_t>(start) : std::floor(start);
```
Needed for consistency, since static_cast<int> rounds towards zero whereas std::floor rounds towards negative.